admin/sections/config/chat.tsx

```diff
@@ -0,0 +1,96 @@
+import { ExpandableSection } from "@/components/ui/custom/expandableSection";
+import { MultiInput } from "@/components/ui/custom/multiInput";
+import { SubmitButton } from "@/components/ui/custom/submitButton";
+import {
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Textarea } from "@/components/ui/textarea";
+import { useEffect, useRef } from "react";
+import { reloadDemoChat } from "../demoChat";
+
+// Get the first not null or undefined value
+function useInitialValue(value: any) {
+  const ref = useRef();
+
+  useEffect(() => {
+    if (value !== null && value !== undefined && ref.current === undefined) {
+      ref.current = value;
+    }
+  }, [value]);
+
+  return ref;
+}
+
+export const ChatConfig = ({
+  form,
+  isSubmitting,
+  addCallback,
+}: {
+  form: any;
+  isSubmitting: boolean;
+  addCallback: any;
+}) => {
+  // To detect the change of conversation starters
+  const conversationStarters = form.watch("conversation_starters");
+  const initialConversationStartersRef = useInitialValue(conversationStarters);
+
+  async function onClickSubmit() {
+    if (
+      JSON.stringify(conversationStarters) !==
+      JSON.stringify(initialConversationStartersRef.current)
+    ) {
+      initialConversationStartersRef.current = conversationStarters;
+      addCallback((prev: any) => [...prev, reloadDemoChat]);
```

Comment:
maybe instead of the callback, set a `fromUpdated` state and pass it to the parent?